wire-protoc-integration/src/main/java/com/squareup/wire/protocwire/ProtocFileSystem.kt

```diff
@@ -0,0 +1,45 @@
+package com.squareup.wire.protocwire
+
+import okio.Buffer
+import okio.ForwardingFileSystem
+import okio.Path
+import okio.Sink
+import okio.Timeout
+
+internal class ProtocFileSystem(private val response: Plugin.Response): ForwardingFileSystem(SYSTEM) {
```

Comment:
Let's add a comment as to why we need this file system, namely that we don't wanna write on disk but do memory manipulation on the response

Reply:
I added a comment: ```/**
 * The primary responsibility of the protoc plugin is to do some memory
 * manipulation and deferring to protoc to manage the creation of the generated
 * files through the [Plugin.Response].
 *
 * The [ProtocFileSystem] helps with replacing Wire's file writing behavior with
 * one that defers to protoc to handle generated files.
 */``` wdyt?